### PR TITLE
Pulls non-package specific documentation repos into Jekyll

### LINKED ISSUE #12: Crawl non-package specific documentation repositories
We'd want non-package specific documentation content (e.g. ROS install instructions) to show up in the rosindex too, preferably at `/doc`. To that end, documentation repositories need to be:

- [ ] Listed in a VCS repos file within rosindex repo.
- [ ] Pulled into Jekyll crawling path using VCS.

Generated content in turn needs to be made accessible through the site.

### PR DESCRIPTION
**Pull request by hidmic**
Closes #12. Support for both Markdown and ReST file formats is provided ~but these must carry [front matter](https://jekyllrb.com/docs/front-matter/) or are otherwise ignored by Jekyll~. Links between files within the documentation repository are also working. Top level repo READMEs are used as index. Documentation repositories are thus kept completely clean and decoupled from Jekyll.

A [dummy repository](https://github.com/hidmic/ros2_documentation) was created solely for testing purposes. We should replace it when real documentation repositories are available.

Connected to [ros-infrastructure/rosindex_containers#2](https://github.com/ros-infrastructure/rosindex_containers/pull/2).

### CODE REVIEW

File: _data/repos/docs.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# VCS repos file listing all non-package specific
+# documentation repositories. Its file format, however,
+# has been extended to include, on a per repository basis,
+# the following extra fields:
```
Comment (apojomovsky) [inline _data/repos/docs.yml:4]: @hidmic do you think it would be worth specifying that the vcs tool will safely ignore the additional fields?

**Review by tfoote — APPROVED** [review]
lgtm it looks like this now has a merge conflict to be resolved before merging.

File: _data/repos/docs.yml (hunk 1 above)
Comment (hidmic) [reply _data/repos/docs.yml:4]: Yes, that makes sense to me.

**Comment by hidmic** [comment]
Alright, PR rebased and comments addressed! Thanks for the review guys. Merging.

**Pull request merged**